### PR TITLE
Provider gradle add verification provider version

### PR DESCRIPTION
**Pull request by toastyblast**
[Issue #929](https://github.com/DiUS/pact-jvm/issues/929) - Added "version" variable to Pact Provider during the pactVerify task in pact-jvm-provider-gradle.
This is to stop pact from using the project.version as its hardcoded value (it's now the default, if no version is given to the serviceProvider).

### CODE REVIEW

**Comment by toastyblast** [comment]
I'm not fully sure if my change in the PR above actually does what I want it to, as I'm not familiar with the codebase (and I can't test it in the project I need it in until it is released in a version I can add as a dependency in Gradle). But if it indeed doesn't work, I at least hope it gets the idea across of what I need.

**Comment by toastyblast** [comment]
Also may I add that when working on this, I couldn't run the full ./gradlew clean build command as suggested in the "How to contribute" page of pact. It will fail on task "run 200 requests to mock server". Any way to fix this so I can run builds properly locally?

Additionally, when trying to do task "signMavenProductionProduction" it will also fail, saying there's no signatory. Any suggestions for this as well?

File: gradlew.bat [hunk 1]
```diff
@@ -14,7 +30,7 @@ set APP_BASE_NAME=%~n0
 set APP_HOME=%DIRNAME%
 
 @rem Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-set DEFAULT_JVM_OPTS=
+set DEFAULT_JVM_OPTS="-Xmx64m" "-Xms64m"
```
Comment (Scot3004) [inline gradlew.bat]: This environment value is put by the CI server, could you please remove it

File: gradlew [hunk 2]
```diff
@@ -28,7 +44,7 @@ APP_NAME="Gradle"
 APP_BASE_NAME=`basename "$0"`
 
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS=""
+DEFAULT_JVM_OPTS='"-Xmx64m" "-Xms64m"'
```
Comment (Scot3004) [inline gradlew]: This environment value is put by the CI server, could you please remove it

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-bin.zip
```
Comment (Scot3004) [inline gradle/wrapper/gradle-wrapper.properties]: Why did you changed this value?

File: gradlew [hunk 4]
```diff
@@ -1,5 +1,21 @@
 #!/usr/bin/env sh
 
+#
```
Comment (Scot3004) [inline gradlew]: i don't think that you should update this file, i not find relationship between your issue and this file

Comment (toastyblast) [reply gradlew]: All issues you mentioned weirdly I haven't changed manually at all. Maybe my IntelliJ set them automatically, or Gradle did. Wouldn't be able to tell you, but I'll remove/change everything you mentioned 👍

**Comment by toastyblast** [comment]
Problem is, when I do "./gradlew clean build" (as instructed by the contribution guide), it still fails 
":consumer:pact-jvm-consumer:test > Executing test au.com.dius.pact.consumer.MockHttpServerSpec > handle more than 200 tests"
due to a "org.spockframework.runtime.SpockTimeoutError at MockHttpServerSpec.groovy:57".

**Comment by uglyog** [comment]
That test has a 60 second timeout on it. It's probably better to have that run on CI only. I'll make the change.

Don't know why you're getting the signMavenProductionProduction error. It should only require a key to sign the JARs when publishing. You can run run `./gradlew build x signMavenProductionProduction` to skip it.

**Comment by uglyog** [comment]
After reviewing this PR, I think the better way is to allow you to override the callback that returns the version instead of introducing a new property.

I.e.,

```
pact {
   serviceProviders {
      'Provider-Name' {
         // Mock server host settings...

         providerVersion = { branchName() + '-' + abbreviatedId() }

         if ('pactVerify' in gradle.startParameter.taskNames) {
            hasPactsFromPactBroker('http://localhost:80/')
         }
      }
   }
}
```

**Comment by toastyblast** [comment]
So, does this already work, or is this just a proposal? I fear I'm not well acquainted enough with the workflow or code of Pact to implement such a thing myself at the moment.

**Comment by uglyog** [comment]
After reviewing the code, the callback exists but it set by the Task, not in the configuration. But I'm going to move the property to the Gradle specific class so it won't effect any other build tools.

I'm also thinking of renaming the publish property to providerVersion so they can't be confused with the project version from Gradle.

**Comment by toastyblast** [comment]
So, after this change was merged a new version was created, however, this version does not include the fix.

Now 15 days later, I still haven't seen any updates. Can I be given any idea or indication of when to expect this change to be implemented so we can make use of it?

**Comment by uglyog** [comment]
It's merged and will be in the next release. I'll organise a release this weekend.

**Comment by toastyblast** [comment]
Thank you for the heads up. I've seen the release, I'll get to implementing it for my team now. Once again, thanks a lot for the help!

**Pull request merged**